external-crates/move/crates/move-vm-profiler/src/lib.rs

```diff
@@ -184,23 +171,42 @@ impl GasProfiler {
     }
 
     pub fn to_file(&self) {
-        if !*PROFILER_ENABLED || !self.is_metered() {
+        let Some(config) = &self.config else {
+            return;
+        };
+        if !self.is_metered() {
+            info!("No meaningful gas usage for this transaction, it may be a system transaction");
             return;
         }
-        // Get the unix timestamp
+
+        let mut p = config.full_path.clone();
+        let file_prefix = match p.file_stem() {
+            Some(fp) => fp.to_str().unwrap_or("gas_profile"),
```

Comment:
all the Osstr library functions that would allow some form of concatenation appear to be under nightly, and
```
 let filename = format!(
            "{:?}_{}_{}.{:?}",
            file_prefix,
            self.name.clone(),
            now,
            file_extension
        );
``` 
leads to the OsStr components getting wrapped in quotation marks, like "output"_d_t."json"